uaa/src/test/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpointDocs.java

```diff
@@ -59,7 +59,7 @@ void info_endpoint_for_json() throws Exception {
                 fieldWithPath("commit_id").type(STRING).description("The GIT sha for the UAA version"),
                 fieldWithPath("timestamp").type(STRING).description("JSON timestamp for the commit of the UAA version"),
                 fieldWithPath("idpDefinitions").optional().type(OBJECT).description("A list of alias/url pairs of SAML IDP providers configured. Each url is the starting point to initiate the authentication process for the SAML identity provider."),
-                fieldWithPath("idpDefinitions.*").optional().type(ARRAY).description("A list of alias/url pairs of SAML IDP providers configured. Each url is the starting point to initiate the authentication process for the SAML identity provider."),
+                fieldWithPath("idpDefinitions.*").optional().type(STRING).description("The URL to initiate the authentication process for the SAML identity provider."),
```

Comment:
ok, it is not a JSON array but it is also not a pure String, but an JSON object simply...

So depends from which side you want solve it. Pure syntax, then object or human readable, then it is more an array

Reply:
Hi, why is `idpDefinitions.*` a JSON object? 
Based on the sample /info endpoint output, it looks like `idpDefinitions`'s value is an object (hence line 61 in the quoted code block is asserting so). But it looks like the value of `idpDefinitions.*` is an URL string. Also see [test here](https://github.com/cloudfoundry/uaa/blob/bef96ed0af5b7cd2cbe27dc495aa5b7cf1a68dec/server/src/test/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpointTests.java#L535-L540) showing that `idpDefinitions` is a `Map<String, String>`.

Reply:
because these Maps are JSON objects. see https://www.w3schools.com/js/js_json_objects.asp

My point is, array maybe was not the best choice, but String is even worse, so either stay with this type or use the correct one, which is Object

Reply:
> because these Maps are JSON objects.
In the `fieldWithPath` syntax here, `idpDefinitions` is the map (key-value pairs), but `idpDefinitions.*` is the key in the key-value pair. 

The line changed in this commit is a test (that will get turned into doc), if we put:
```
fieldWithPath("idpDefinitions.*").optional().type(OBJECT).description("The URL to initiate the authentication process for the SAML identity provider.")
```
or 
```
fieldWithPath("idpDefinitions.*").optional().type(ARRAY).description("The URL to initiate the authentication process for the SAML identity provider.")
```
This test will fail, once you configure one or more SAML IDP in [its UAA.yml](https://github.com/cloudfoundry/uaa/blob/11b086c4017ba6b107f92ff42d17f66faf16080d/uaa/src/test/resources/integration_test_properties.yml#L95).

Reply:
ok, understood, you had a test failure if you fill the yml...
Then I propose VARIES. It generates this map, which to me is no STRING

"idpDefinitions" : {
    "dummy" : "http://localhost:8080/uaa/saml/discovery?returnIDParam=idp&entityID=cloudfoundry-saml-login&idp=dummy&isPassive=true",
    "xs2security" : "http://localhost:8080/uaa/saml/discovery?returnIDParam=idp&entityID=cloudfoundry-saml-login&idp=xs2security&isPassive=true"
  },

Reply:
I think we still have an misunderstanding of the `fieldWithPath` syntax. `idpDefinitions`'s value is 100% a map, NOT a string. That's why the test `fieldWithPath("idpDefinitions").optional().type(OBJECT).description("xxx")` is passing. 

But `fieldWithPath("idpDefinitions.*")` refers NOT to `idpDefinitions`'s value, but `idpDefinitions.dummy`'s value, where the `*` is matching `dummy`. So `idpDefinitions.*`'s value is really "http://localhost:8080/uaa/saml/discovery?returnIDParam=idp&entityID=cloudfoundry-saml-login&idp=dummy&isPassive=true".

Reply:
I know and I have run the tests with content you suggested and I could verify the test failures, however with VARIES the tests were succeed with OK

Another solution could be to have 2 extra entries, means instead idpDefinitions.*
* idpDefinitions.alias
* idpDefinitions.url 

both with type string .